Modifying the HelpText for AzRecoveryServiceBackup cmdlets

hiaga opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Tuned the help text for AzRecoveryServiceBackup cmdlets.

## Checklist

- [Yes ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ Yes] The title of the PR is clear and informative
- [Yes] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [Yes] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [N/A] If applicable, the changes made in the PR have proper test coverage
- [N/A ] For public API changes to cmdlets:
    - [N/A ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [N/A] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

VeryEarly commented on src/RecoveryServices/RecoveryServices/help/Register-AzRecoveryServicesBackupContainer.md:
```diff
@@ -27,7 +27,7 @@ Register-AzRecoveryServicesBackupContainer [-Container] <ContainerBase>
 
 ## DESCRIPTION
-The cmdlet registers an Azure VM for AzureWorkloads with specific workloadType.
+This command allows Azure Backup to convert the �Resource� to a �Backup Container� which is then registered to the given Recovery services vault. The Azure Backup service can then discover workloads of the given workload type within this container to be protected later.
```
please fix the special characters

VeryEarly commented on src/RecoveryServices/RecoveryServices.Backup/Cmdlets/Container/GetAzureRmRecoveryServicesBackupContainer.cs:
```diff
@@ -41,7 +41,7 @@ public class GetAzureRmRecoveryServicesBackupContainer : RSBackupVaultCmdletBase
             HelpMessage = ParamHelpMsgs.Container.BackupManagementType)]
         [ValidateNotNullOrEmpty]
-        [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage")]
+        [ValidateSet("AzureVM", "MARS", "AzureWorkload", "AzureStorage")]
```
this looks like breaking change to me, you can keep AzureSql and add AzureWorkload

VeryEarly commented:
/azp run

VeryEarly commented:
Hi @hiaga ,

Saw you reverted AzureWorkload, is it not needed anymore?

you can still keep it by adding AzureSQL back:  [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage", "AzureWorkload")]

hiaga commented:
> Hi @hiaga ,
> 
> Saw you reverted AzureWorkload, is it not needed anymore?
> 
> you can still keep it by adding AzureSQL back: [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage", "AzureWorkload")]

Hi @VeryEarly, yes AzureWorkload is needed and added. Pls review.

VeryEarly commented:
> > Hi @hiaga ,
> > Saw you reverted AzureWorkload, is it not needed anymore?
> > you can still keep it by adding AzureSQL back: [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage", "AzureWorkload")]
> 
> Hi @VeryEarly, yes AzureWorkload is needed but we have a small issue. Currently, all the AzRecoveryServicesBackup cmdlets support backup of SQL DBs via BackupManagementType = "AzureWorkload" and WorkloadType = "MSSQL". Previously similar use-case was supported via BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase", so basically this is no longer supported and is causing confusion for our customers (as they should ideally use AzureWorkload+ MSSQL combination). Now that is the reason we want to remove it from help text and every other place, so we want to understand why would it be a concern and treated as a breaking change when we remove it from ValidationSet (it is anyway not supported and validation set has to be ideally updated to AzureWorkload)?
> 
> and currently, I have added AzureWorkload in the validation set as it is the way we currently support backup of SQL DBs.

For example, If customers already had used "AzureSQL", removed it from validation would failed their existing script.

However, just checked the source code, has this method: https://github.com/Azure/azure-powershell/blob/master/src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs#L503

ever been implemented? If not, that means  BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase" combination never worked right?

If that's the case, I think it's fine to remove AzureSQL from validation set because it won't work anyway.

pvrk commented:
> > > Hi @hiaga ,
> > > Saw you reverted AzureWorkload, is it not needed anymore?
> > > you can still keep it by adding AzureSQL back: [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage", "AzureWorkload")]
> > 
> > 
> > Hi @VeryEarly, yes AzureWorkload is needed but we have a small issue. Currently, all the AzRecoveryServicesBackup cmdlets support backup of SQL DBs via BackupManagementType = "AzureWorkload" and WorkloadType = "MSSQL". Previously similar use-case was supported via BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase", so basically this is no longer supported and is causing confusion for our customers (as they should ideally use AzureWorkload+ MSSQL combination). Now that is the reason we want to remove it from help text and every other place, so we want to understand why would it be a concern and treated as a breaking change when we remove it from ValidationSet (it is anyway not supported and validation set has to be ideally updated to AzureWorkload)?
> > and currently, I have added AzureWorkload in the validation set as it is the way we currently support backup of SQL DBs.
> 
> For example, If customers already had used "AzureSQL", removed it from validation would failed their existing script.
> 
> However, just checked the source code, has this method: https://github.com/Azure/azure-powershell/blob/master/src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs#L503
> 
> ever been implemented? If not, that means BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase" combination never worked right?
> 
> If that's the case, I think it's fine to remove AzureSQL from validation set because it won't work anyway.

@VeryEarly : It was implemented earlier but we stopped the support for this combination more than a year ago. As of now, customers can't enable backup for this combination (AzureSQL + AzureSQLDatabase). So, I hope we can remove this.

VeryEarly commented:
> > > > Hi @hiaga ,
> > > > Saw you reverted AzureWorkload, is it not needed anymore?
> > > > you can still keep it by adding AzureSQL back: [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage", "AzureWorkload")]
> > > 
> > > 
> > > Hi @VeryEarly, yes AzureWorkload is needed but we have a small issue. Currently, all the AzRecoveryServicesBackup cmdlets support backup of SQL DBs via BackupManagementType = "AzureWorkload" and WorkloadType = "MSSQL". Previously similar use-case was supported via BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase", so basically this is no longer supported and is causing confusion for our customers (as they should ideally use AzureWorkload+ MSSQL combination). Now that is the reason we want to remove it from help text and every other place, so we want to understand why would it be a concern and treated as a breaking change when we remove it from ValidationSet (it is anyway not supported and validation set has to be ideally updated to AzureWorkload)?
> > > and currently, I have added AzureWorkload in the validation set as it is the way we currently support backup of SQL DBs.
> > 
> > 
> > For example, If customers already had used "AzureSQL", removed it from validation would failed their existing script.
> > However, just checked the source code, has this method: https://github.com/Azure/azure-powershell/blob/master/src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs#L503
> > ever been implemented? If not, that means BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase" combination never worked right?
> > If that's the case, I think it's fine to remove AzureSQL from validation set because it won't work anyway.
> 
> @VeryEarly : It was implemented earlier but we stopped the support for this combination more than a year ago. As of now, customers can't enable backup for this combination (AzureSQL + AzureSQLDatabase). So, I hope we can remove this.

Make sense, @hiaga you can go ahead remove the AzureSQL from the validation set, sorry I didn't dig into the implementation deep enough before.

hiaga commented:
> > > > > Hi @hiaga ,
> > > > > Saw you reverted AzureWorkload, is it not needed anymore?
> > > > > you can still keep it by adding AzureSQL back: [ValidateSet("AzureVM", "MARS", "AzureSQL", "AzureStorage", "AzureWorkload")]
> > > > 
> > > > 
> > > > Hi @VeryEarly, yes AzureWorkload is needed but we have a small issue. Currently, all the AzRecoveryServicesBackup cmdlets support backup of SQL DBs via BackupManagementType = "AzureWorkload" and WorkloadType = "MSSQL". Previously similar use-case was supported via BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase", so basically this is no longer supported and is causing confusion for our customers (as they should ideally use AzureWorkload+ MSSQL combination). Now that is the reason we want to remove it from help text and every other place, so we want to understand why would it be a concern and treated as a breaking change when we remove it from ValidationSet (it is anyway not supported and validation set has to be ideally updated to AzureWorkload)?
> > > > and currently, I have added AzureWorkload in the validation set as it is the way we currently support backup of SQL DBs.
> > > 
> > > 
> > > For example, If customers already had used "AzureSQL", removed it from validation would failed their existing script.
> > > However, just checked the source code, has this method: https://github.com/Azure/azure-powershell/blob/master/src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/AzureSqlPsBackupProvider.cs#L503
> > > ever been implemented? If not, that means BackupManagementType = "AzureSQL" and WorkloadType = "AzureSQLDatabase" combination never worked right?
> > > If that's the case, I think it's fine to remove AzureSQL from validation set because it won't work anyway.
> > 
> > 
> > @VeryEarly : It was implemented earlier but we stopped the support for this combination more than a year ago. As of now, customers can't enable backup for this combination (AzureSQL + AzureSQLDatabase). So, I hope we can remove this.
> 
> Make sense, @hiaga you can go ahead remove the AzureSQL from the validation set, sorry I didn't dig into the implementation deep enough before.

@VeryEarly - removing AzureSQL, please review and approve.

hiaga commented:
Hi @veryEarly, As per we discussed above I have removed AzureSQL from validationSet and it being considered a breaking change by the static analysis tool. I hope this won't be a blocker again, please review and let me know if any concerns. Thanks.

VeryEarly commented:
Hi @hiaga ,

Please copy BreakingChangeIssues.csv 
from https://dev.azure.com/azure-sdk/public/_build/results?buildId=442396&view=artifacts&type=publishedArtifacts  

to https://github.com/MabOneSdk/azure-powershell/tree/helpTextUpdates/tools/StaticAnalysis/Exceptions/Az.RecoveryServices

hiaga commented:
Hi @veryEarly, added the suggested file. Please review

VeryEarly commented:
/azp run azure-powershell - windows-powershell
